src/main/java/com/ibm/etcd/client/EtcdClient.java

```diff
@@ -338,32 +336,47 @@ public void close() {
             try {
                 channel.shutdown().awaitTermination(2, SECONDS);
             } catch (InterruptedException e) {}
-            executeWhenIdle(() -> internalExecutor.shutdownGracefully(0, 1, SECONDS));
-        });
+            executeWhenIdle(() -> internalExecutor.shutdownGracefully(0, 1, SECONDS), false);
+        }, true);
     }
     
-    // used only in clean shutdown logic
-    final Iterable<SingleThreadEventLoop> eventLoops;
-    
     /**
```

Comment:
it would be better to update this description including `yield`

Reply:
Thanks @yuzheng21. Actually I just pushed a commit which simplifies this to remove the `yield` param altogether - the false case isn't really needed.